p3/data/_validation.py

```diff
@@ -1,49 +1,48 @@
-# Copyright (c) 2022-2023 Intel Corporation
-# SPDX-License-Identifier: MIT
-
 import json
 import jsonschema
-import pkgutil
-
+import os
 
-def _validate_coverage_json(json_string: str) -> object:
+def _validate_coverage_data(coverage_data) -> object:
     """
-    Validate coverage JSON string against schema.
+    Validate coverage data (either JSON string or object) against schema.
 
     Parameters
     ----------
-    json_string : String
-        The JSON string to validate.
+    coverage_data : String or Object
+        The JSON string or object to validate.
 
     Returns
     -------
     Object
-        A Python object corresponding to the JSON string.
+        A Python object corresponding to the JSON data.
 
     Raises
     ------
     ValueError
-        If the JSON string fails to validate.
+        If the data fails to validate.
 
     TypeError
-        If the JSON string is not a string.
+        If the data is not a string or dictionary.
     """
-    if not isinstance(json_string, str):
-        raise TypeError("Coverage must be a JSON string.")
-
-    instance = json.loads(json_string)
-
-    schema_string = pkgutil.get_data(__name__, "coverage-0.3.0.schema")
-    if not schema_string:
+    if isinstance(coverage_data, str):
+        instance = json.loads(coverage_data)
+    elif isinstance(coverage_data, dict):
+        instance = coverage_data
+    else:
+        raise TypeError("Coverage must be a JSON string or dictionary.")
```

Comment:
JSON isn't necessarily a `dict`:
```suggestion
    elif isinstance(coverage_data, (dict, list)):
        instance = coverage_data
    else:
        raise TypeError("Coverage must be a JSON string or object.")
```